src/Compilers/CSharp/Portable/Binder/Binder_Invocation.cs

```diff
@@ -1300,11 +1308,12 @@ private BoundExpression GetDefaultParameterSpecialNoConversion(SyntaxNode syntax
 
             // In a scenario like `string Prop { get; } = M();`, the containing symbol could be the synthesized field.
             // We want to use the associated user-declared symbol instead where possible.
-            var containingMember = ContainingMember() switch
+            var containingMember = InAttributeArgument ? attributedMember : ContainingMember() switch
             {
                 FieldSymbol { AssociatedSymbol: { } symbol } => symbol,
                 var c => c
             };
+            Debug.Assert(InAttributeArgument || (attributedMember is null && containingMember is not null));
```

Comment:
If `InAttributeArgument` is true, do we want to assert a non-null `attributedMember`?

Reply:
Currently, we wouldn't be able to honor such an assertion in semantic model scenarios.

Reply:
Also, I think it should be null if the attribute is applied to something that is not a member, such as an assembly.